ui/tests/integration/components/config-ui/messages/page/create-and-edit-message-test.js

```diff
@@ -0,0 +1,136 @@
+    this.context = { owner: this.engine };
+    this.store = this.owner.lookup('service:store');
+    this.message = this.store.createRecord('config-ui/message');
+  });
+
+  test('it should display all the create form fields and default radio button values', async function (assert) {
+    await render(
+      hbs`<Messages::Page::CreateAndEditMessageForm @message={{this.message}} @authenticated={{true}} />`,
+      {
+        owner: this.engine,
+      }
+    );
+
+    assert.dom('[data-test-page-title]').hasText('Create message');
+    assert
+      .dom('[data-test-form-subtext]')
+      .hasText('Create a custom message for all users when they access a Vault system via the UI.');
+    assert.dom(PAGE.radioGroup('authenticated')).exists();
+    assert.dom(`${PAGE.radioGroup('authenticated')} ${PAGE.radioField('authenticated')}`).isChecked();
+    assert.dom(PAGE.radioGroup('type')).exists();
+    assert.dom(`${PAGE.radioGroup('type')} ${PAGE.radioField('banner')}`).isChecked();
+    assert.dom(PAGE.field('title')).exists();
+    assert.dom(PAGE.field('message')).exists();
+    assert.dom(PAGE.field('linkTitle')).exists();
+    assert.dom(PAGE.field('linkHref')).exists();
+    assert.dom(PAGE.input('startTime')).exists();
+    assert
+      .dom(PAGE.input('startTime'))
+      .hasValue(format(addDays(startOfDay(new Date()), 1), localDateTimeString));
+    assert.dom(PAGE.input('endTime')).exists();
+    assert.dom(PAGE.input('endTime')).hasValue('');
+  });
+
+  test('it should create new message', async function (assert) {
+    assert.expect(1);
+
+    this.server.post('/sys/config/ui/custom-messages', () => {
+      assert.ok(true, 'POST request made to create message');
+    });
+
+    await render(
+      hbs`<Messages::Page::CreateAndEditMessageForm @message={{this.message}} @authenticated={{true}} />`,
+      {
+        owner: this.engine,
+      }
+    );
+    await fillIn(PAGE.input('title'), 'Awesome custom message title');
+    await fillIn(
+      PAGE.input('message'),
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Pulvinar mattis nunc sed blandit libero volutpat sed cras ornare.'
+    );
+    await fillIn(
+      PAGE.input('startTime'),
+      format(addDays(startOfDay(new Date('2023-12-12')), 1), localDateTimeString)
+    );
+    await click('#specific-date');
+    await fillIn(
+      PAGE.input('endTime'),
+      format(addDays(startOfDay(new Date('2023-12-12')), 10), localDateTimeString)
+    );
+
+    await click(PAGE.button('create-message'));
+  });
+
+  test('it should show a disabled create form button when there are no title or message', async function (assert) {
+    await render(
+      hbs`<Messages::Page::CreateAndEditMessageForm @message={{this.message}} @authenticated={{true}} />`,
+      {
+        owner: this.engine,
+      }
+    );
+    assert.dom(PAGE.button('create-message')).isDisabled();
+  });
```

Comment:
Might be good to check that the validation messages render too.

Reply:
I was thinking about that and what validations to check for! Since everything is defaulted to a specific values and the button doesn't show until the required fields are added in, I'm not sure 🤔

Reply:
Since the button was changed using the form errors check, the user will now see these form vaildations if they click the button so I added tests for that